usvm-ts/src/test/kotlin/org/usvm/util/TSMethodTestRunner.kt

```diff
@@ -91,9 +91,10 @@ open class TSMethodTestRunner : TestRunner<TSTest, MethodDescriptor, EtsType?, T
 
     override val typeTransformer: (Any?) -> EtsType
         get() = {
-            require(it is KClass<*>) { "Only TSObjects are allowed" }
+            // Both KClass and TSObject instances come here
+            val temp = if (it is KClass<*> || it == null) it else it::class
```

Comment:
How is it possible? Isn't it a mistake? How a TSObject could come here if the signature for it is

```
  protected inline fun <reified T : TSObject, reified R : TSObject> 
  ...
  arrayOf(typeTransformer(T::class), typeTransformer(R::class))
```
?

Reply:
typeTransformer is used in parent class _TestRunner_ in method _checkType_:
<pre>
private fun checkTypes(
        argumentsTypes: Array&lt;out Type&gt;,
        valuesToCheck: List&lt;List&lt;Any?&gt;&gt;,
    ) {
        for (valueToCheck in valuesToCheck) {
            val actualArgsTypes = valueToCheck.map { <b>typeTransformer</b>(it) }
            val expectedTypesWithActual = argumentsTypes.zip(actualArgsTypes)

           ...
        }
    }
</pre>

Which results in TSObject entity coming in TSMethodTestRunner _typeTransfomer_. Running any sample without this check will result in this error:

```java.lang.IllegalStateException: Should not be called: class org.usvm.TSObject$TSNumber$Double```